VideoApi/VideoApi.Services/Handlers/Base/CustomSelfTestApiDelegatingHandler.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Threading;
+using System.Threading.Tasks;
+using VideoApi.Common.Security.Supplier.Base;
+namespace VideoApi.Services.Handlers.Base;
+
+public abstract class CustomSelfTestApiDelegatingHandler : DelegatingHandler
+{
+    private readonly ICustomJwtTokenProvider _tokenProvider;
+
+    protected CustomSelfTestApiDelegatingHandler(ICustomJwtTokenProvider tokenProvider)
+    {
+        _tokenProvider = tokenProvider;
+    }
+
+    private const string ParticipantIdName = "participantId";
+
+    protected override Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
+    {
+        var participantId = request.Properties.ContainsKey(ParticipantIdName) 
+            ? request.Properties[ParticipantIdName] 
+            : throw new Exception($"Could not find the field {ParticipantIdName} in the request properties dictionary");
```

Comment:
could we maybe throw a `KeyNotFoundException` here instead? That should fix the SonarCloud issue